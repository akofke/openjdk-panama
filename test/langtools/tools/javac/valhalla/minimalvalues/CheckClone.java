/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support clone
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckClone.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror -Xlint:values CheckClone.java
 */
@jdk.incubator.mvt.ValueCapableClass
final class CheckClone {
    @jdk.incubator.mvt.ValueCapableClass
    final class InnerValue {
        void foo(InnerValue iv) {
            iv.clone(); // <-- error
        }
    }
    @Override
    protected Object clone() { return null; } // <-- error
}
