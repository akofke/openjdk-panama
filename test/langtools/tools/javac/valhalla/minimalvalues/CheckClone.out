- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckClone.java:12:21: compiler.warn.value.does.not.support: calling java.lang.Object's method: clone
CheckClone.java:16:22: compiler.warn.value.does.not.support: overriding java.lang.Object's method: clone
1 error
3 warnings
